src/dotty/tools/dotc/transform/SyntheticMethods.scala

```diff
@@ -118,9 +123,22 @@ class SyntheticMethods extends MiniPhaseTransform with IdentityDenotTransformer
 
     /** The class
      *
+     *  class C(x: T) extends AnyVal
+     *
+     *  gets the hashCode method:
+     *
+     *    def hashCode: Int = x.hashCode()
+     */
+    def valueHashCodeBody(implicit ctx: Context): Tree = {
+      assert(accessors.length == 1)
+      ref(accessors.head).select(nme.hashCode_).appliedToNone
```

Comment:
I used `appliedToNone` to avoid errors like:
```
type mismatch:
 found   : ()Int($this.a.hashCode)
 required: Int
```
But this breaks `dotc_core` with:
```
error: method hashCode in class TermName does not take parameters
```
Any idea what the proper way to fix this is?